pmd-apex/src/main/resources/category/apex/design.xml

```diff
@@ -86,6 +86,22 @@ public class Complicated {
         </example>
     </rule>
 
+    <rule name="CognitiveComplexity"
+          message="The {0} ''{1}'' has a{2} cognitive complexity of {3}."
+          since="6.0.0"
+          class="net.sourceforge.pmd.lang.apex.rule.design.CognitiveComplexityRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_apex_design.html#cognitivecomplexity">
+        <description>
+            TODO
```

Comment:
What you enter here will appear in the generated online documentation -> Example entry for cyclomatic complexity: https://pmd.github.io/latest/pmd_rules_apex_design.html#cyclomaticcomplexity

The description should help users to decide, whether they want to enable this rule or not. In this case, it should probably shortly explain what cognitive complexity is and link to the paper. You may also reason about the advantages of simpler software (less bugs, easier to understand, less maintenance cost).

The example section below could give 2 or 3 short examples to explain, how the metric is calculated - similar to what you did in the tests (`// +1` or `// +2`).

Reply:
Please check what I added for the existing Apex rules